### ISSUE TITLE
Issue when using borderSide and mask

### ISSUE THREAD

**Issue by tbaranes** [IBAnimatable/IBAnimatable#225]
While trying to reproduce #224, I run in this bug:

<img width="149" alt="screen shot 2016-06-17 at 22 20 10" src="https://cloud.githubusercontent.com/assets/1402212/16163467/ab3cb422-34d9-11e6-967a-ebcd1b13f75e.png">

In order to reproduce it, just add the following line in the playground:

```
view.borderSide = "Top,Bottom"
```

Seems that we are not resetting as expected the mask when using a custom `borderSide`

**Comment by JakeLin** [comment]
@tbaranes good find, the problem is caused by  

```
  private func maskCircle() {
    layer.cornerRadius = ceil(min(bounds.width, bounds.height))/2
  }
```

I think we should use `layer.mask` instead of `cornerRadius` to mask a circle.

**Comment by tbaranes** [comment]
Just played a bit much with `borderSide` and `maskType`, it seems there a second bug related to them: `borderSide` isn't applied if we are using a custom `mask`. It applied the default border (all), and apply the custom sides only if no mask settled or if we are using the circle one (probably for the reason you said before).

Definitely something wrong  🤔

**Comment by JakeLin** [comment]
@tbaranes I think that's by design (at least my idea 😁),  the `borderSide` property is similar to `border-width` in CSS. Please have a look at https://developer.mozilla.org/en/docs/Web/CSS/border-width. In CSS we can use `border-top-width` or `border-right-width` and so on to specify one side of the border. It is very useful for Material Desgin apps. But it can't support masked shape, and it doesn't make sense to support non-rectangle shapes. I think we can fix the circle one and add a document comment for `borderSide` to let the user know it only support rectangle (a view with custom mask or cornerRadius). How do you think this approach?

**Comment by tbaranes** [comment]
I'm still finding that approach strange, but if it's a design choice, I'm ok with it. As you said, it doesn't make sense to support non-rectangle shapes with specific border size. If someone ask for that feature we concrete example, it will always be time to improve the result.

However, that definitely needs documentations 👍

**Comment by JakeLin** [comment]
OK, I will work on the fix and the documentations. thanks @tbaranes

**Comment by JakeLin** [comment]
Fix in #234
